docs/proposals/llm-d.md

```diff
@@ -81,6 +84,43 @@ Our current Northstar designs lay out the initial scope (join llm-d-contributors
 * [Prefix Cache Hierarchy](https://docs.google.com/document/d/1inTneLEZTv3rDEBB9KLOB9K6oMq8c3jkogARJqdt_58/edit)
 * [Variant Autoscaling over Hardware, Workload, and Traffic](https://docs.google.com/document/d/1inTneLEZTv3rDEBB9KLOB9K6oMq8c3jkogARJqdt_58/edit)
 
+### Components
+
+llm-d streamlines deployment and integration of the following components:
+
+1. An **inference scheduler** directing traffic to model servers
+    1. Integrates with the Kubernetes [inference gateway API project](https://gateway-api-inference-extension.sigs.k8s.io/) to have Kubernetes control plane API
+    2. Performing model routing and rollout, flow control, kv- and prefix- cache aware load balancing
+    3. Balances traffic to the optimal model server based on the request, workload type, and current load
+2. **vLLM model servers** deployed onto Kubernetes
+    1. In single host or multi-host (using [LeaderWorkerSets](https://lws.sigs.k8s.io/) and Ray as best practice) configurations
+    2. With native support for disaggregated serving and optional curated plugins for advanced features
+    3. Using project recommended defaults or highly customized user settings
+    4. May be deployed in multiple deployment **variants** (hardware, software, topology) that offer different performance tradeoffs
+3. vLLM default prefix caching and zero or more **prefix cache integrations**
+    1. At least one remote prefix cache option
+    2. At least one prefix cache option using the local SSD drives for each replica efficiently
+4. A **variant autoscaler** working with the inference scheduler and Kubernetes horizontal pod autoscaling
```

Comment:
should this be generalized to cover all use cases related autoscaling. As I understand it, the variant autoscaler concern is scaling Ps and Ds for a given *model*. We also want to include autoscaling across models, minimizing downtime due to frequent scaling up and down, optimizing costs and meet SLAs.

Reply:
The scope for variant autoscaler is heterogenous workload and traffic scaling as per the north star to rebalance capacity availability vs capacity demand (due to current traffic mix).

I do not know what autoscaling across models means in this context - can you elaborate.

Reply:
thanks for clarifying the scope, that's basically what I meant by "across model".

My second item, how quickly vllm instances are reconfigured/created is covered by item 2.5.